server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java

```diff
@@ -0,0 +1,123 @@
+      new SumFormula(CLASSES_KEY),
+      new SumFormula(FUNCTIONS_KEY),
+      new SumFormula(STATEMENTS_KEY),
+      new SumFormula(PUBLIC_API_KEY),
+      new SumFormula(COMMENT_LINES_KEY),
+      new SumFormula(PUBLIC_UNDOCUMENTED_API_KEY),
+      new SumFormula(COMPLEXITY_KEY),
+      new SumFormula(COMPLEXITY_IN_CLASSES_KEY),
+      new SumFormula(COMPLEXITY_IN_FUNCTIONS_KEY),
+      // new SumFormula(LINES_TO_COVER_KEY),
+      // new SumFormula(NEW_LINES_TO_COVER_KEY),
+      // new SumFormula(UNCOVERED_LINES_KEY),
+      // new SumFormula(NEW_UNCOVERED_LINES_KEY),
+      // new SumFormula(CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(NEW_CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(NEW_UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(IT_LINES_TO_COVER_KEY),
+      // new SumFormula(NEW_IT_LINES_TO_COVER_KEY),
+      // new SumFormula(IT_UNCOVERED_LINES_KEY),
+      // new SumFormula(NEW_IT_UNCOVERED_LINES_KEY),
+      // new SumFormula(IT_CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(NEW_IT_CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(IT_UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(NEW_IT_UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(OVERALL_LINES_TO_COVER_KEY),
+      // new SumFormula(NEW_OVERALL_LINES_TO_COVER_KEY),
+      // new SumFormula(OVERALL_UNCOVERED_LINES_KEY),
+      // new SumFormula(NEW_OVERALL_UNCOVERED_LINES_KEY),
+      // new SumFormula(OVERALL_CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(NEW_OVERALL_CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(OVERALL_UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(NEW_OVERALL_UNCOVERED_CONDITIONS_KEY),
+
+      // Distribution formulas
+      new DistributionFormula(FUNCTION_COMPLEXITY_DISTRIBUTION_KEY),
+      new DistributionFormula(FILE_COMPLEXITY_DISTRIBUTION_KEY),
+
+      // Average formulas, must be executed after all sum formulas as they depend on their measures
+      AverageFormula.Builder.newBuilder().setOutputMetricKey(FILE_COMPLEXITY_KEY)
+        .setMainMetricKey(COMPLEXITY_KEY)
+        .setByMetricKey(FILES_KEY)
+        .build(),
+      AverageFormula.Builder.newBuilder().setOutputMetricKey(CLASS_COMPLEXITY_KEY)
+        .setMainMetricKey(COMPLEXITY_IN_CLASSES_KEY)
+        .setByMetricKey(CLASSES_KEY)
+        .setFallbackMetricKey(COMPLEXITY_KEY)
+        .build(),
+      AverageFormula.Builder.newBuilder().setOutputMetricKey(FUNCTION_COMPLEXITY_KEY)
+        .setMainMetricKey(COMPLEXITY_IN_FUNCTIONS_KEY)
+        .setByMetricKey(FUNCTIONS_KEY)
+        .setFallbackMetricKey(COMPLEXITY_KEY)
+        .build()
+      );
+    FORMULAS = formulaList.build();
+  }
+
+  /**
+   * Return list of formulas that was previously provided by CoreMetrics
+   *
+   * TODO When all decorators will be moved to CE, uncomment commented lines to activate all formulas and remove formulas declaration in {@link org.sonar.api.measures.CoreMetrics}
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)